inc/Modules/Rest/Workflow_Controller.php

```diff
@@ -354,226 +330,226 @@ public function execute_plugin_action( \WP_REST_Request $request ): \WP_REST_Res
 		$plugin_type      = $decoded_body['plugin_type'] ?? 'public';
 		$plugin_path_info = $decoded_body['plugin_path_info'] ?? '';
 
-		if ( ! in_array( $action, array( 'activate', 'deactivate', 'update', 'remove', 'change-version', 'install' ), true ) ) {
-			return new \WP_Error( 'invalid_action', __( 'Invalid action provided.', 'oneupdate' ), array( 'status' => 400 ) );
+		if ( ! in_array( $action, [ 'activate', 'deactivate', 'update', 'remove', 'change-version', 'install' ], true ) ) {
+			return new \WP_Error( 'invalid_action', __( 'Invalid action provided.', 'oneupdate' ), [ 'status' => 400 ] );
 		}
 
 		if ( empty( $slug ) || ! is_string( $slug ) ) {
-			return new \WP_Error( 'invalid_slug', __( 'Invalid plugin slug provided.', 'oneupdate' ), array( 'status' => 400 ) );
+			return new \WP_Error( 'invalid_slug', __( 'Invalid plugin slug provided.', 'oneupdate' ), [ 'status' => 400 ] );
 		}
 
 		if ( ! is_array( $sites ) || empty( $sites ) ) {
-			return new \WP_Error( 'invalid_sites', __( 'Invalid sites provided.', 'oneupdate' ), array( 'status' => 400 ) );
+			return new \WP_Error( 'invalid_sites', __( 'Invalid sites provided.', 'oneupdate' ), [ 'status' => 400 ] );
 		}
 
-		$output = array();
-		$errors = array();
+		$output = [];
+		$errors = [];
 		if ( 'activate' === $action || 'deactivate' === $action || 'remove' === $action ) {
 			foreach ( $sites as $site ) {
-				$oneupdate_sites = $GLOBALS['oneupdate_sites'] ?? array();
-				$api_key         = $oneupdate_sites[ $site ]['apiKey'] ?? '';
+				$oneupdate_sites = Settings::get_shared_sites();
+				$api_key         = $oneupdate_sites[ $site ]['api_key'] ?? '';
 
 				$request_postfix = '/wp-json/' . self::NAMESPACE . '/oneupdate-plugins-options';
 				// strip the trailing slash from the site URL.
-				$site_url = rtrim( $oneupdate_sites[ $site ]['siteUrl'], '/' );
-
-				if ( ! empty( $api_key ) ) {
-					$response = wp_remote_post(
-						$site_url . $request_postfix,
-						array(
-							'headers' => array(
-								'Content-Type' => 'application/json',
-								'X-OneUpdate-Plugins-Token' => $api_key,
-							),
-							'body'    => wp_json_encode(
-								array(
-									'options' => array(
-										'plugins'     => array( $plugin_path_info ),
-										'plugin_type' => $action,
-									),
-								)
-							),
-							'timeout' => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout -- this is to avoid timeout issues.
-						)
+				$site_url = rtrim( $oneupdate_sites[ $site ]['url'], '/' );
+
+				if ( empty( $api_key ) ) {
+					continue;
+				}
+
+				$response = wp_remote_post(
+					$site_url . $request_postfix,
+					[
+						'headers' => [
+							'Content-Type'      => 'application/json',
+							'X-OneUpdate-Token' => $api_key,
+						],
+						'body'    => wp_json_encode(
+							[
+								'options' => [
+									'plugins'     => [ $plugin_path_info ],
+									'plugin_type' => $action,
+								],
+							]
+						),
```

Comment:
wp_json_encode() can return `false`

```suggestion
						) ?: '',
```

Reply:
Done at every instance.